sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs

```diff
@@ -31,7 +31,7 @@ internal class AzureMonitorTransmitter : ITransmitter
         internal readonly TransmitFromStorageHandler? _transmitFromStorageHandler;
         private bool _disposed;
 
-        public AzureMonitorTransmitter(AzureMonitorExporterOptions options, IPlatform platform)
+        public AzureMonitorTransmitter(AzureMonitorExporterOptions options, IPlatform platform, IVmMetadataProvider vmMetadataProvider)
```

Comment:
Shall we combine these interfaces `IPlatform` and `IVmMetadataProvider`?

Reply:
I don't have a strong opinion here, either approach would be valid.

I kept them separate because AzureVmMetadata is an external service, and not a Platform detail. 
And I like the separate smaller implementations. :)